### PR TITLE
cache fullSetupRequired locally

### PR DESCRIPTION
**Pull request by icewind1991**
no need to hit redis every time

Signed-off-by: Robin Appelman <robin@icewind.nl>

### CODE REVIEW

File: lib/private/Files/SetupManager.php [hunk 1]
```diff
@@ -434,7 +435,10 @@ private function fullSetupRequired(IUser $user): bool {
 		// we perform a "cached" setup only after having done the full setup recently
 		// this is also used to trigger a full setup after handling events that are likely
 		// to change the available mounts
-		return !$this->cache->get($user->getUID());
+		if (!isset($this->fullSetupRequired[$user->getUID()])) {
+			$this->fullSetupRequired[$user->getUID()] = !$this->cache->get($user->getUID());
```
Comment (juliushaertl) [inline lib/private/Files/SetupManager.php:439]:
```suggestion
			$this->fullSetupRequired[$user->getUID()] = $this->cache->get($user->getUID());
```

Isn't that inverted then on the return statement again?

Comment (icewind1991) [reply lib/private/Files/SetupManager.php:439]: good catch, forgot to remove the invert on return

**Review by juliusknorr — APPROVED** [review]

**Review by nickvergessen — APPROVED** [review]

**Review by nickvergessen — APPROVED** [review]

**Pull request merged**